server/providers/challenges/rdeploy-blob/index.js

```diff
@@ -0,0 +1,52 @@
+const config = require('../../../../config/server')
+const path = require('path')
+const fs = require('fs')
+
+class RDeployBlobProvider {
+  constructor ({ options, onUpdate }) {
+    options = {
+      updateInterval: 60 * 1000,
+      useGlobalRDeployDirectory: true,
+      // rDeployDirectory: ''
+      ...options
+    }
+
+    this._onUpdate = onUpdate
```

Comment:
Instead of only supporting one update listener, the plugin could emit events on an EventEmitter.